src/exercise/06.js

```diff
@@ -4,25 +4,21 @@
 import * as React from 'react'
 
 function UsernameForm({onSubmitUsername}) {
-  // 🐨 add a submit event handler here (`handleSubmit`).
-  // 💰 Make sure to accept the `event` as an argument and call
-  // `event.preventDefault()` to prevent the default behavior of form submit
-  // events (which refreshes the page).
-  //
-  // 🐨 get the value from the username input (using whichever method
-  // you prefer from the options mentioned in the instructions)
-  // 💰 For example: event.target.elements[0].value
-  // 🐨 Call `onSubmitUsername` with the value of the input
+  const [username, setUsername] = React.useState(null)
+  const usernameRef = React.useRef(null);
 
-  // 🐨 add the onSubmit handler to the <form> below
+  const handleChange = (event) => {
+    setUsername(event.target.value.toLowerCase());
+  }
 
-  // 🐨 make sure to associate the label to the input.
-  // to do so, set the value of 'htmlFor' prop of the label to the id of input
   return (
-    <form>
+    <form onSubmit={(e) => {
+        e.preventDefault();
+        onSubmitUsername(usernameRef.current.value);
+      }}>
```

Comment:
I'm not sure if I like this more or less than a separate callback function. 😄 It's definitely encapsulated!

Reply:
Not going to lie, I wanted so badly to convert this one to a class-based component. I hate having nested functions. Coming from PHP, it just feels wrong.

That said, in this case I prefer this format just because all the wrapping function does is prevent default. I like how Laravel Livewire handles things, with a `submit.prevent` hook that does the prevent default without me having to.